lib/neo4j/spatial.rb

```diff
@@ -1,143 +1,180 @@
+
         options = {
-          layer: layer,
-          lat: lat || 'lat',
-          lon: lon || 'lon'
+          name: name,
+          type: type || 'point',
+          encoderConfig: "#{lon || 'lon'}:#{lat || 'lat'}"
         }
+        wrap_spatial_procedure('addLayer', options)
+      end
 
-        spatial_post('/ext/SpatialPlugin/graphdb/addSimplePointLayer', options)
+      def remove_layer(name)
+        options = {name: name}
+        wrap_spatial_procedure('removeLayer', options, node: false)
       end
 
-      def add_editable_layer(layer, format = 'WKT', node_property_name = 'wkt')
+      def add_point_layer(layer)
+        options = {layer: layer}
+
+        wrap_spatial_procedure('addPointLayer', options)
+      end
+
+      def add_wkt_layer(layer, node_property_name = 'wkt')
         options = {
           layer: layer,
-          format: format,
-          nodePropertyName: node_property_name
+          node_property_name: node_property_name
         }
 
-        spatial_post('/ext/SpatialPlugin/graphdb/addEditableLayer', options)
+        wrap_spatial_procedure('addWKTLayer', options)
       end
 
-      def get_layer(layer)
-        options = {
-          layer: layer
-        }
-        spatial_post('/ext/SpatialPlugin/graphdb/getLayer', options)
+      def get_layer(layer, execute: true)
+        options = {layer: layer}
+        wrap_spatial_procedure('layer', options, execute: execute)
       end
 
-      def add_geometry_to_layer(layer, geometry)
+      def add_wkt(layer, geometry, execute: true)
         options = {
           layer: layer,
           geometry: geometry
         }
-        spatial_post('/ext/SpatialPlugin/graphdb/addGeometryWKTToLayer', options)
+        wrap_spatial_procedure('addWKT', options, execute: execute)
       end
 
-      def edit_geometry_from_layer(layer, geometry, node)
+      def update_from_wkt(layer, geometry, node, execute: true)
         options = {
           layer: layer,
           geometry: geometry,
           geometryNodeId: get_id(node)
         }
-        spatial_post('/ext/SpatialPlugin/graphdb/updateGeometryFromWKT', options)
+        wrap_spatial_procedure('updateFromWKT', options, execute: execute)
       end
 
-      def add_node_to_layer(layer, node)
+      # Hmmm this one has trouble, because we actually need to MATCH the node itself...
+      # Wish this could be cleaner but for now it works...
+      def add_node_to_layer(layer, node, execute: true)
         options = {
           layer: layer,
-          node: "#{resource_url}node/#{node.neo_id}"
+          node_id: node.neo_id
         }
-        spatial_post('/ext/SpatialPlugin/graphdb/addNodeToLayer', options)
+
+        query_ = Query.new(session: self)
+        procedure = query_.match(:n)
+                    .where('id(n) = {node_id}')
+                    .with(:n).call('spatial.addNode({layer}, n) YIELD node')
+                    .return('node')
+                    .params(options)
```

Comment:
I would call this variable `params` or `parameters` instead of `options` because `options` is generally for a hash which affects the behavior of a method.  Also, you could do:

```ruby
procedure = query_.match(:n)
            .where(n: {neo_id: node.neo_id})
            .with(:n).call('spatial.addNode({layer}, n) YIELD node').params(layer: layer)
            .return(:node)
```